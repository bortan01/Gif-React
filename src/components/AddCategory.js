import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
   const [inputValue, setInputValue] = useState("");
   const handleInputChance = (e) => {
      setInputValue(e.target.value);
   }
   const handleSubmit = (e) => {
      e.preventDefault();
      console.log("submit");
      if (inputValue.trim().length > 2) {
         setCategories((categorias) => [inputValue, ...categorias]);
         setInputValue('');
      }
   }
   return (
      <form onSubmit={handleSubmit}>
         <input
            type="text"
            value={inputValue}
            onChange={(e) => { handleInputChance(e) }}
         />
      </form>
   )
}

AddCategory.propTypes = {
   setCategories: PropTypes.func.isRequired
}