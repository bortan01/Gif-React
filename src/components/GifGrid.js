import React from 'react'

export const GifGrid = ({category}) => {

   const getGifs =async ()=>{
      const url ='https://api.giphy.com/v1/gifs/search?q=goku&limit=10&api_key=R4qK3NSI73RXvjZSdZ8hCDYXSfEq0xSj';
      const respuesta = await fetch(url);
      const {data} = await respuesta.json();

      const gifs = data.map((img)=>{
         return {
            id : img.id,
            url : img.images?.downsized_medium.url,
            title : img.title
         }
      });
      console.log(gifs);
   }
   getGifs();
   return (
      <div>
         <h3>{category}</h3>
      </div>
   )
}
