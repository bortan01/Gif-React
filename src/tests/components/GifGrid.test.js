import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFechGifs } from "../../hooks/useFechGifs";
// import moduleName from '@testing-library/jest-dom/dist/extend-expect';
jest.mock('../../hooks/useFechGifs');
describe('pruebas en <GifGrid/>', () => {
   const category = 'luffy';

   test('debe de mostrar items cuando se cargan imagenes useFechGifs sin datos', () => {
      useFechGifs.mockReturnValue({
         data: [],
         loading: true
      });
      const wraper = shallow(<GifGrid category={category} />);
      expect(wraper).toMatchSnapshot();
   });
   test('debe de mostrar items cuando se cargan imagenes useFechGifs con datos', () => {
      const gifs = [{
         id: "ABC-1234",
         url: "https//localhost/imagen.jpg",
         title: "Imagen XD"
      }];
      //ESTA ES UNA LLAMADA FICTICIA CON SU RESPECTIVA RESPUESTA 
      useFechGifs.mockReturnValue({
         data: gifs,
         loading: false
      });
      const wraper = shallow(<GifGrid category={category} />);
      expect(wraper).toMatchSnapshot();
      expect(wraper.find('p').exists()).toBe(false);
      expect(wraper.find('GifGridItem').length).toBe(gifs.length);
   });




});
