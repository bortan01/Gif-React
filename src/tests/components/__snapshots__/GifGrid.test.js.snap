// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid/> debe de mostrar items cuando se cargan imagenes useFechGifs con datos 1`] = `
<Fragment>
  <h3>
    luffy
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC-1234"
      key="ABC-1234"
      title="Imagen XD"
      url="https//localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en <GifGrid/> debe de mostrar items cuando se cargan imagenes useFechGifs sin datos 1`] = `
<Fragment>
  <h3>
    luffy
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
